File: apps/mobile/lib/core/network/api_response.dart
/// 统一 API 响应结构
class ApiResponse<T> {
  /// 响应码（200 表示成功）
  final int code;

  /// 响应消息
  final String msg;

  /// 响应数据
  final T? data;

  ApiResponse({
    required this.code,
    this.msg = '',
    this.data,
  });

  /// 是否成功
  bool get isSuccess => code == 200;

  /// 是否失败
  bool get isFailure => !isSuccess;

  /// 工厂方法：从 JSON 解析
  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Map<String, dynamic>)? fromJsonT,
  ) {
    return ApiResponse(
      code: json['code'] as int? ?? 200,
      msg: json['msg'] as String? ?? '',
      data: json['data'] != null && fromJsonT != null
          ? fromJsonT(json['data'] as Map<String, dynamic>)
          : json['data'] as T?,
    );
  }

  /// 转换为 JSON
  Map<String, dynamic> toJson() {
    return {
      'code': code,
      'msg': msg,
      'data': data,
    };
  }

  @override
  String toString() {
    return 'ApiResponse(code: $code, msg: $msg, data: $data)';
  }
}

File: apps/server/src/main/java/com/chainpass/vo/ApiResponse.java
package com.chainpass.vo;

import java.io.Serializable;

/**
 * 统一API响应结构
 */
public class ApiResponse<T> implements Serializable {

    private Integer code;
    private String msg;
    private T data;

    // Getters and Setters
    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    // Static factory methods
    public static <T> ApiResponse<T> success() {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(200);
        return response;
    }

    public static <T> ApiResponse<T> success(T data) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(200);
        response.setData(data);
        return response;
    }

    public static <T> ApiResponse<T> success(String msg, T data) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(200);
        response.setMsg(msg);
        response.setData(data);
        return response;
    }

    public static <T> ApiResponse<T> error(String msg) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(400);
        response.setMsg(msg);
        return response;
    }

    public static <T> ApiResponse<T> error(Integer code, String msg) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(code);
        response.setMsg(msg);
        return response;
    }
}